watertap/unit_models/tests/test_dewatering_unit.py

```diff
@@ -0,0 +1,276 @@
+            17216.2434 * units.mg / units.liter
+        )
+        m.fs.unit.inlet.conc_mass_comp[0, "X_S"].fix(2611.4843 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BH"].fix(1e-6 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_BA"].fix(1e-6 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_P"].fix(626.0652 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_O"].fix(1e-6 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NO"].fix(1e-6 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "S_NH"].fix(
+            1442.7882 * units.mg / units.liter
+        )
+        m.fs.unit.inlet.conc_mass_comp[0, "S_ND"].fix(0.54323 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ND"].fix(100.8668 * units.mg / units.liter)
+        m.fs.unit.inlet.alkalinity.fix(97.8459 * units.mol / units.m**3)
+
+        return m
+
+    @pytest.mark.build
+    @pytest.mark.unit
+    def test_build(self, du):
+
+        assert hasattr(du.fs.unit, "inlet")
+        assert len(du.fs.unit.inlet.vars) == 5
+        assert hasattr(du.fs.unit.inlet, "flow_vol")
+        assert hasattr(du.fs.unit.inlet, "conc_mass_comp")
+        assert hasattr(du.fs.unit.inlet, "temperature")
+        assert hasattr(du.fs.unit.inlet, "pressure")
+        assert hasattr(du.fs.unit.inlet, "alkalinity")
+
+        assert hasattr(du.fs.unit, "underflow")
+        assert len(du.fs.unit.underflow.vars) == 5
+        assert hasattr(du.fs.unit.underflow, "flow_vol")
+        assert hasattr(du.fs.unit.underflow, "conc_mass_comp")
+        assert hasattr(du.fs.unit.underflow, "temperature")
+        assert hasattr(du.fs.unit.underflow, "pressure")
+        assert hasattr(du.fs.unit.underflow, "alkalinity")
+
+        assert hasattr(du.fs.unit, "overflow")
+        assert len(du.fs.unit.overflow.vars) == 5
+        assert hasattr(du.fs.unit.overflow, "flow_vol")
+        assert hasattr(du.fs.unit.overflow, "conc_mass_comp")
+        assert hasattr(du.fs.unit.overflow, "temperature")
+        assert hasattr(du.fs.unit.overflow, "pressure")
+        assert hasattr(du.fs.unit.overflow, "alkalinity")
+
+        assert number_variables(du) == 76
+        assert number_total_constraints(du) == 60
+        assert number_unused_variables(du) == 0
+
+    @pytest.mark.unit
+    def test_dof(self, du):
+        assert degrees_of_freedom(du) == 0
+
+    @pytest.mark.unit
+    def test_units(self, du):
+        assert_units_consistent(du)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, du):
+
+        iscale.calculate_scaling_factors(du)
+        initialization_tester(du)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, du):
```

Comment:
There should typically be a test to check for badly scaled variables here. I checked, and a few will crop up. We can be lenient about it and make a note to revisit in our master issue, considering the whole effort here for the BSM2 flowsheet, or you could add the typical test to check for badly scaled vars and eliminate any that exist by adding scaling. Though the badly scaled vars are probably attributed to zero components, it's hard to say whether this will end up impacting model stability at the flowsheet level later on.

Reply:
Actually, through our off-GH discussion, you made it clear that the badly scaled vars stem from the lack of scaling on the ASM1 property model, which we have noted in our master issue (#934)